### PR TITLE
V1 API: Support api changes, cleanup msg wait/search api 

### PR DESCRIPTION
**Pull request by magik6k**
On top of https://github.com/filecoin-project/lotus/pull/5685

This PR:
* Makes the changes needed to be made across the codebase to actually support different API versions
  * CLI uses v0 api by default
  * The rest of the codebase should be using v1
* Collapses `State[Wait/Search]MsgLimited,StateGetReceipt` into `State[Wait/Search]Msg` whth added params which should make it much harder to use incorrectly

### CODE REVIEW

File: api/wrap.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package api
+
+import (
+	"reflect"
+)
+
+// Wrap adapts partial api impl to another version
+// proxyT is the proxy type used as input in wrapperT
+// Usage: Wrap(new(v1api.FullNodeStruct), new(v0api.WrapperV1Full), eventsApi).(EventAPI)
+func Wrap(proxyT, wrapperT, impl interface{}) interface{} {
```
Comment (Kubuxu) [inline api/wrap.go:10]: What does the `impl` do?

Comment (magik6k) [reply api/wrap.go:10]: It provides a partial implementation of the interface, which is used to populate the struct

**Review by Kubuxu — COMMENTED** [review]
SGWM

**Review by arajasek — APPROVED** [review]
Ship it!

**Comment by magik6k** [comment]
(merging into the v1 api branch, so things are in one place)

**Pull request merged**